AutomatedTesting/Gem/PythonTests/Atom/atom_utils/material_canvas_utils.py

```diff
@@ -8,103 +8,109 @@
+        self.name = atomtools.GraphDocumentRequestBus(
+            bus.Event, GraphDocumentRequestBusEvents.GET_GRAPH_NAME, self.document_id)
+        self.graph_object = atomtools.GraphDocumentRequestBus(
+            bus.Event, GraphDocumentRequestBusEvents.GET_GRAPH, self.document_id)
+        self.nodes = []
+
+    def _create_node_by_name(self, node_name: str) -> object:
+        """
+        Creates a new node in memory matching the node_name string on the specified graph object.
+        i.e. "World Position" for node_name would create a World Position node.
+        :param node_name: String representing the type of node to add to the graph.
+        :return: azlmbr.object.PythonProxyObject representing a C++ AZStd::shared_ptr<Node> object.
+        """
+        return atomtools.DynamicNodeManagerRequestBus(
+            bus.Broadcast, DynamicNodeManagerRequestBusEvents.CREATE_NODE_BY_NAME, self.graph_object, node_name)
+
+    def add_node(self, node_name: str, position: math.Vector2) -> None:
+        """
+        Creates a node in memory using _create_node_by_name then adds it to this Graph object.
+        :param node_name: String representing the type of node to add to the graph.
+        :param position: math.Vector2(x,y) value that determines where to place the node on the Graph's grid.
+        :return: None, but appends the new node to self.nodes list.
+        """
+        node_object = self._create_node_by_name(node_name)
+        node_id = graph.GraphControllerRequestBus(
+            bus.Event, GraphControllerRequestBusEvents.ADD_NODE, self.graph_id, node_object, position)
+        if not node_id:
+            return  # If the node isn't valid, don't add it.
+        self.nodes.append(Node(node_object, node_id, node_name, self.graph_id, position))
+
+
+class Slot(object):
+    """
+    Represents a slot that is part of a Node (required).
+    """
+
+    def __init__(self, slot_name):
+        self.slot_name = slot_name
+        self.id = graph.GraphModelSlotId(slot_name)
+        if not self.id:
+            self.id = None
+
+
+class Node(object):
+    """
+    Represents a node inside of a Graph (required).
+    """
+
+    def __init__(self, node_object: object, node_id: int, node_name: str, graph_id: math.Uuid, position: math.Vector2):
+        self.node_object = node_object
+        self.node_id = node_id
+        self.node_name = node_name
+        self.graph_id = graph_id
+        self.position = position
+        self.slots = []
+
+    def add_slot(self, slot_name: str) -> None:
+        """
+        Adds a new Slot object to this Node to identify one of its existing slots.
+        :param slot_name: Name of the slot to lookup using graph.GraphModelSlotId().
+        :return: None, but adds the new slot to the self.slots list.
+        """
+        self.slots.append(Slot(slot_name))
+
+    def connect_slots(self, source_slot: Slot, target_node: (), target_slot: Slot) -> azlmbr.object.PythonProxyObject:
```

Comment:
This would be cleaner if it was moved to the graph class and mirrored the same parameters as the graph controller request bus